sematic/api/endpoints/runs.py

```diff
@@ -137,6 +138,11 @@ def list_runs_endpoint(user: Optional[User]) -> flask.Response:
         if sql_predicates is not None:
             query = query.filter(sql_predicates)
 
+        search_string = flask.request.args.get("search")
+        search_predicates = _generate_search_predicate(search_string)
+        if search_predicates is not None:
+            query = query.filter(search_predicates)
```

Comment:
Do it this way:
```suggestion
        search_string = flask.request.args.get("search")
        if search_string is not None:
            search_predicates = _generate_search_predicate(search_string)
            query = query.filter(search_predicates)
```

That way, `_generate_search_predicate` can have `search_string: str` as input and `BooleanClauseList` as output, no `Optional`.
It is much cleaner in terms of contract, and simplifies the flow (only one `if` in total instead of two).

Reply:
Good call. Thanks